### PR TITLE
State manage api calls

### PR DESCRIPTION
**Pull request by kotesh-arya**
## List of features included in this PR

-  Post and edit feature on every post
-  Profile Page with edit feature
-  Like, comment, bookmark, follow, unfollow and feed features.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -2,17 +2,65 @@ import "./App.css";
...
+          path="/explore"
+          element={
+            <RequiresAuth>
+              <Explore />
+            </RequiresAuth>
+          }
+        />
+        <Route
+          path="/bookmarks"
+          element={
```
Comment (Dev-AkshatGupta) [inline src/App.js:42]: You can separate the whole route component.

File: src/Components/CommentContainer.jsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  const dispatch = useDispatch();
+  return (
+    <Box
+      bg={bgColor}
+      display={"flex"}
+      flexDirection={"row"}
+      justifyContent={"flex-start"}
+      alignItems={"flex-start"}
+      marginBottom={"15px"}
+      // border={"2px solid red"}
```
Comment (Dev-AkshatGupta) [inline src/Components/CommentContainer.jsx:37]: remove this comment

File: src/Components/CommentContainer.jsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    >
+      <Box display={"flex"} alignItems={"center"} flexDirection={"row"}>
+        <Avatar src="" name={username} marginRight={"15px"} />
+      </Box>
+      <Box
+        display={"flex"}
+        //  border={"2px solid red"}
+        // bg={"red"}
```
Comment (Dev-AkshatGupta) [inline src/Components/CommentContainer.jsx:45]: remove these comments

File: src/Components/EditCommentModal.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+              </Box>
+            </form>
+          </ModalBody>
+
+          {/* <ModalFooter> */}
+          {/* <Button colorScheme="blue" mr={3} onClick={onClose}>
+          Close
+        </Button> */}
+          {/* <Button variant="ghost">Secondary Action</Button> */}
+          {/* </ModalFooter> */}
```
Comment (Dev-AkshatGupta) [inline src/Components/EditCommentModal.js:91]: Remove the commented code.

File: src/Components/EditPostModal.jsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+            </form>
+          </ModalBody>
+
+          {/* <ModalFooter> */}
+          {/* <Button colorScheme="blue" mr={3} onClick={onClose}>
+            Close
+          </Button> */}
+          {/* <Button variant="ghost">Secondary Action</Button> */}
+          {/* </ModalFooter> */}
+        </ModalContent>
```
Comment (Dev-AkshatGupta) [inline src/Components/EditPostModal.jsx:91]: Remove the commented code. Or either add TODO

File: src/Components/EditUserModal.jsx [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+                </Box>
+              </Box>
+            </form>
+          </ModalBody>
+
+          <ModalFooter>
+            {/* <Button colorScheme="blue" mr={3} onClick={onClose}>
+              Close
+            </Button> */}
+            {/* <Button variant="ghost">Secondary Action</Button> */}
```
Comment (Dev-AkshatGupta) [inline src/Components/EditUserModal.jsx:146]: Remove commented code.

File: src/Components/NewPostModal.jsx [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+import { FaPen } from "react-icons/fa";
+import { RiImageAddLine } from "react-icons/ri";
+import { useDispatch, useSelector } from "react-redux";
+import { createPost, getAllPosts } from "../features/post/postsSlice";
+
+function NewPostModal() {
+  const dispatch = useDispatch();
+  const { user, token } = useSelector((store) => store.auth);
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  // const [postContent, setPostContent] = useState("");
```
Comment (Dev-AkshatGupta) [inline src/Components/NewPostModal.jsx:28]: Remove commented code.

File: src/Components/NewPostModal.jsx [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+                </Box>
+              </Box>
+            </form>
+          </ModalBody>
+
+          {/* <ModalFooter> */}
+          {/* <Button colorScheme="blue" mr={3} onClick={onClose}>
+            Close
+          </Button> */}
+          {/* <Button variant="ghost">Secondary Action</Button> */}
```
Comment (Dev-AkshatGupta) [inline src/Components/NewPostModal.jsx:99]: Remove the commented code.

**Review by Dev-AkshatGupta — APPROVED** [review]
Reviewed the code. Please have look at them and else everything looks good.

**Review by Namrata-J — APPROVED** [review]

**Review by swethaogeti — COMMENTED** [review]
good to merge !!

**Review by swethaogeti — APPROVED** [review]

File: src/Components/NewPostModal.jsx (hunk 7 above)
Comment (kotesh-arya) [reply src/Components/NewPostModal.jsx:28]: Sure

File: src/App.js (hunk 1 above)
Comment (kotesh-arya) [reply src/App.js:42]: Sure

**Pull request merged**